File: trunk/theLink/example/csharp/Filter2.cs
/**
 *  \file       example/csharp/Filter2.cs
 *  \brief      \$Id: Filter2.cs 507 2009-11-28 15:18:46Z dev1usr $
 *  
 *  (C) 2009 - NHI - #1 - Project - Group
 *  
 *  \version    \$Rev: 507 $
 *  \author     EMail: aotto1968 at users.sourceforge.net
 *  \attention  this software has GPL permissions to copy
 *              please contact AUTHORS for additional information
 */
using System;
using csmsgque;

namespace example {
  sealed class Filter2 : MqS, IFilterFTR {

    // service definition from IFilterFTR
    void IFilterFTR.Call () {
      throw new ApplicationException("my error");
    }

    public static void Main(String[] argv) {
      Filter2 srv = new Filter2();
      try {
	srv.ConfigSetName("filter");
	srv.LinkCreate(argv); 
	srv.ProcessEvent(MqS.WAIT.FOREVER);
      } catch (Exception ex) {
        srv.ErrorSet (ex);
      }
      srv.Exit();
    }
  }
}

File: trunk/theLink/example/vb/Filter2.vb
REM
REM  \file       theLink/example/vb/Filter2.vb
REM  \brief      \$Id$
REM  
REM  (C) 2009 - NHI - #1 - Project - Group
REM  
REM  \version    \$Rev$
REM  \author     EMail: aotto1968 at users.berlios.de
REM  \attention  this software has GPL permissions to copy
REM              please contact AUTHORS for additional information
REM


Imports System
Imports csmsgque

Public Module example
  Private Class Filter2
    Inherits MqS
    
    ' constructor
    Public Sub New(ByVal tmpl As MqS)
      MyBase.New(tmpl)
      ConfigSetIsServer(True)
    End Sub

    ' service definition
    Public Sub FilterFTR()
      Throw New ApplicationException("my error")
    End Sub

  End Class

  Sub Main(ByVal args() As String)
    Dim srv As Filter2 = MqFactoryS(Of Filter2).Add("filter").[New]()
    Try
      srv.ConfigSetName("filter")
      srv.ConfigSetIsServer(True)
      srv.LinkCreate(args)
      srv.ServiceCreate("+FTR", AddressOf srv.FilterFTR)
      srv.ServiceProxy("+EOF")
      srv.ProcessEvent(MqS.WAIT.FOREVER)
    Catch ex As Exception
      srv.ErrorSet(ex)
    Finally
      srv.Exit()
    End Try
  End Sub
End Module

